exchange/seat_non_bids_test.go

```diff
@@ -102,9 +72,300 @@ var sampleSeatBids = func(seat string, nonBidCount int) []openrtb_ext.SeatNonBid
+		toAppend []SeatNonBidBuilder
+		expected SeatNonBidBuilder
+	}{
+		{
+			name:     "nil receiver",
+			builder:  nil,
+			toAppend: []SeatNonBidBuilder{{"seat1": []openrtb_ext.NonBid{{ImpId: "imp1"}}}},
+			expected: nil,
+		},
+		{
+			name:     "empty builder",
+			builder:  SeatNonBidBuilder{},
+			toAppend: []SeatNonBidBuilder{{"seat1": []openrtb_ext.NonBid{{ImpId: "imp1"}}}},
+			expected: SeatNonBidBuilder{"seat1": []openrtb_ext.NonBid{{ImpId: "imp1"}}},
+		},
+		{
+			name:     "multiple seats",
+			builder:  SeatNonBidBuilder{"seat1": []openrtb_ext.NonBid{{ImpId: "imp1"}}},
+			toAppend: []SeatNonBidBuilder{{"seat2": []openrtb_ext.NonBid{{ImpId: "imp2"}}}},
+			expected: SeatNonBidBuilder{"seat1": []openrtb_ext.NonBid{{ImpId: "imp1"}}, "seat2": []openrtb_ext.NonBid{{ImpId: "imp2"}}},
+		},
+		{
+			name:     "multiple appends",
+			builder:  SeatNonBidBuilder{"seat1": []openrtb_ext.NonBid{{ImpId: "imp1"}}},
+			toAppend: []SeatNonBidBuilder{{"seat2": []openrtb_ext.NonBid{{ImpId: "imp2"}}}, {"seat3": []openrtb_ext.NonBid{{ImpId: "imp3"}}}},
+			expected: SeatNonBidBuilder{"seat1": []openrtb_ext.NonBid{{ImpId: "imp1"}}, "seat2": []openrtb_ext.NonBid{{ImpId: "imp2"}}, "seat3": []openrtb_ext.NonBid{{ImpId: "imp3"}}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.builder.append(tt.toAppend...)
+			assert.Equal(t, tt.expected, tt.builder)
+		})
+	}
+}
+
+func TestRejectImps(t *testing.T) {
+	tests := []struct {
+		name    string
+		impIDs  []string
+		builder SeatNonBidBuilder
+		want    SeatNonBidBuilder
+	}{
+		{
+			name:    "nil_imps",
+			impIDs:  nil,
+			builder: SeatNonBidBuilder{},
+			want:    SeatNonBidBuilder{},
+		},
+		{
+			name:    "empty_imps",
+			impIDs:  []string{},
+			builder: SeatNonBidBuilder{},
+			want:    SeatNonBidBuilder{},
+		},
+		{
+			name:    "one_imp",
+			impIDs:  []string{"imp1"},
+			builder: SeatNonBidBuilder{},
+			want: SeatNonBidBuilder{
+				"seat1": []openrtb_ext.NonBid{
+					{
+						ImpId:      "imp1",
+						StatusCode: 300,
+					},
+				},
+			},
+		},
+		{
+			name:    "many_imps",
+			impIDs:  []string{"imp1", "imp2"},
+			builder: SeatNonBidBuilder{},
+			want: SeatNonBidBuilder{
+				"seat1": []openrtb_ext.NonBid{
+					{
+						ImpId:      "imp1",
+						StatusCode: 300,
+					},
+					{
+						ImpId:      "imp2",
+						StatusCode: 300,
+					},
+				},
+			},
+		},
+		{
+			name:   "many_imps_appended_to_prepopulated_list",
+			impIDs: []string{"imp1", "imp2"},
+			builder: SeatNonBidBuilder{
+				"seat0": []openrtb_ext.NonBid{
+					{
+						ImpId:      "imp0",
+						StatusCode: 0,
+					},
+				},
+			},
+			want: SeatNonBidBuilder{
+				"seat0": []openrtb_ext.NonBid{
+					{
+						ImpId:      "imp0",
+						StatusCode: 0,
+					},
+				},
+				"seat1": []openrtb_ext.NonBid{
+					{
+						ImpId:      "imp1",
+						StatusCode: 300,
+					},
+					{
+						ImpId:      "imp2",
+						StatusCode: 300,
+					},
+				},
+			},
+		},
```

Comment:
Can you add the following additional test case:
```
        {
			name:   "many_imps_appended_to_prepopulated_list_same_seat",
			impIDs: []string{"imp1", "imp2"},
			builder: SeatNonBidBuilder{
				"seat1": []openrtb_ext.NonBid{
					{
						ImpId:      "imp0",
						StatusCode: 300,
					},
				},
			},
			want: SeatNonBidBuilder{
				"seat1": []openrtb_ext.NonBid{
					{
						ImpId:      "imp0",
						StatusCode: 300,
					},
					{
						ImpId:      "imp1",
						StatusCode: 300,
					},
					{
						ImpId:      "imp2",
						StatusCode: 300,
					},
				},
			},
		},
```

Reply:
Done